src/app_service/service/contacts/service.nim

```diff
@@ -395,9 +392,9 @@ QtObject:
     if self.contacts.hasKey(publicKey):
       if self.contacts[publicKey].dto.added and not self.contacts[publicKey].dto.removed and contact.added and not contact.removed:
         signal = SIGNAL_CONTACT_UPDATED
-    if contact.removed:
-      singletonInstance.globalEvents.showContactRemoved("Contact removed", fmt "You removed {contact.displayName} as a contact", contact.id)
-      signal = SIGNAL_CONTACT_REMOVED
+      if contact.removed and not self.contacts[publicKey].dto.removed:
```

Comment:
This indentation change of the whole block... intended?

Reply:
Yes. I needed to get access to the old contact (`self.contacts[publicKey]`) and I needed to be in the `hasKey` block.

Anyway, it makes sense, since we should never warn about a contact being removed if we see them for the first time.